secrets/azure: Dynamic Roles known issue (#2643)
hashicorp/web-unified-docs: content/vault/global/partials/important-changes/known-issue/azure-dynamic-roles.mdx

<!-- BEGIN: Vault:=v2.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------
| Known issue | Closed   | Enterprise    | 2.0.0            | 2.0.3

<!-- END: Vault:=v2.x -->

<!-- BEGIN: Vault:=v1.21.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------
| Known issue | Closed   | Enterprise    | 1.21.0           | 1.21.8

<!-- END: Vault:=v1.21.x -->

<!-- BEGIN: Vault:=v1.20.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------
| Known issue | Closed   | Enterprise    | 1.20.0           | 1.20.13

<!-- END: Vault:=v1.20.x -->

<!-- BEGIN: Vault:=v1.19.x -->

| Change      | Status   | Vault edition | Affected version | Fixed version
| ----------- | -------- | ------------- | ---------------- | -------------
| Known issue | Closed   | Enterprise    | 1.19.0           | 1.19.19

<!-- END: Vault:=v1.19.x -->

#### Issue

> Azure is eventually consistent, so internal state propagation may lag after an object is created or modified. When Vault returns a credential for a dynamic role, it must create an application, instantiate a service principal and then provision a new credential. This operation may intermittently fail if the service principal is in an inconsistent state with an error message resembling:

Reviewer: ```suggestion
Azure is eventually consistent, so internal state propagation may lag when Vault
creates or modifies an object. When Vault returns a credential for a dynamic
role, it must create an application, instantiate a service principal, and then
provision a new credential.

When the service principal is in an inconsistent state, Vault may intermittently
return an error similar to the following when the credential requests fails:
```
Style correction: write in active voice, avoid "this" as a pronoun

> ```

Reviewer: ````suggestion
```text
````

The service principal cannot be created, updated, or restored because the service principal name {guid} is already in use.
```

#### Workaround

> The Azure secrets engine now makes use of more efficient API calls to better avoid inconsistent states. It is recommended to upgrade to the latest patch release of Vault in order to resolve the issue.

Reviewer: ```suggestion
Ugrade to latest patch release of Vault. As of the latest Vault versions, the
Azure secrets engine uses of more efficient API calls to better avoid
inconsistent states.
```
Style correction: write in active voice. Make recommendations declaratively
